Source/Scene/ModelInstanceCollection.js

```diff
@@ -0,0 +1,671 @@
+
+            if (dynamic) {
+                Matrix4.multiplyTransformation(viewMatrix, modelMatrix, instanceMatrix);
+            } else {
+                instanceMatrix = modelMatrix;
+            }
+
+            // First three rows of the model matrix
+            typedArray[i * 16 + 0] = instanceMatrix[0];
+            typedArray[i * 16 + 1] = instanceMatrix[4];
+            typedArray[i * 16 + 2] = instanceMatrix[8];
+            typedArray[i * 16 + 3] = instanceMatrix[12] - center.x;
+            typedArray[i * 16 + 4] = instanceMatrix[1];
+            typedArray[i * 16 + 5] = instanceMatrix[5];
+            typedArray[i * 16 + 6] = instanceMatrix[9];
+            typedArray[i * 16 + 7] = instanceMatrix[13] - center.y;
+            typedArray[i * 16 + 8] = instanceMatrix[2];
+            typedArray[i * 16 + 9] = instanceMatrix[6];
+            typedArray[i * 16 + 10] = instanceMatrix[10];
+            typedArray[i * 16 + 11] = instanceMatrix[14] - center.z;
+
+            // Other instance data like pickColor, color, etc. Colors can be packed like in BillboardCollection as needed.
+            typedArray[i * 16 + 12] = pickColor.red;
+            typedArray[i * 16 + 13] = pickColor.green;
+            typedArray[i * 16 + 14] = pickColor.blue;
+            typedArray[i * 16 + 15] = pickColor.alpha;
+        }
+
+        if (createVertexBuffer) {
+            var usage = dynamic ? BufferUsage.STREAM_DRAW : BufferUsage.STATIC_DRAW;
+            vertexBuffer = Buffer.createVertexBuffer({
+                context : context,
+                typedArray : typedArray,
+                usage : usage
+            });
+            collection._vertexBuffer = vertexBuffer;
+            collection._typedArray = typedArray;
+            collection._createVertexBuffer = false;
+        } else {
+            vertexBuffer.copyFromArrayView(typedArray);
+        }
+    }
+
+    function updateBoundingSphere(collection) {
+        var points = [];
+        var instancesLength = collection.instancesLength;
+        for (var i = 0; i < instancesLength; i++) {
+            var translation = new Cartesian3();
+            Matrix4.getTranslation(collection._instances[i].modelMatrix, translation);
+            points.push(translation);
+        }
+
+        var boundingSphere = collection._boundingSphere;
+        BoundingSphere.fromPoints(points, boundingSphere);
+        Matrix4.fromTranslation(boundingSphere.center, collection._boundingSphereModel);
+    }
+
+    function createModel(collection, context) {
+        var i;
+        var instancingSupported = collection._instancingSupported;
+        var modelOptions = {
+            url : collection._url,
+            cacheKey : collection._cacheKey,
+            headers : collection._headers,
+            asynchronous : collection._asynchronous,
+            debugShowBoundingVolume : collection._debugShowBoundingVolume,
```

Comment:
`debugShowBoundingVolume` and `debugWireframe` can change per frame.  Don't set them here.  Set them per frame in `update`.